File: GeneratedEntityReaderClasses/Components/LooseGroundComponent.cs
namespace GeneratedNoitaClasses;
public class LooseGroundComponent : NoitaComponentBase,  iNoitaType<LooseGroundComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaFloat probability { get; set; } = new();
	public NoitaInt max_durability { get; set; } = new();
	public NoitaFloat max_distance { get; set; } = new();
	public NoitaFloat max_angle { get; set; } = new();
	public NoitaInt min_radius { get; set; } = new();
	public NoitaInt max_radius { get; set; } = new();
	public NoitaFloat chunk_probability { get; set; } = new();
	public NoitaFloat chunk_max_angle { get; set; } = new();
	public NoitaInt chunk_count { get; set; } = new();
	public NoitaInt chunk_material { get; set; } = new();
	public NoitaString collapse_images { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		probability.Read(s);
		max_durability.Read(s);
		max_distance.Read(s);
		max_angle.Read(s);
		min_radius.Read(s);
		max_radius.Read(s);
		chunk_probability.Read(s);
		chunk_max_angle.Read(s);
		chunk_count.Read(s);
		chunk_material.Read(s);
		collapse_images.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		probability.Write(s);
		max_durability.Write(s);
		max_distance.Write(s);
		max_angle.Write(s);
		min_radius.Write(s);
		max_radius.Write(s);
		chunk_probability.Write(s);
		chunk_max_angle.Write(s);
		chunk_count.Write(s);
		chunk_material.Write(s);
		collapse_images.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/LooseGroundComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class LooseGroundComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    probability:NoitaFloat = field(default_factory=NoitaFloat)
    max_durability:NoitaInt = field(default_factory=NoitaInt)
    max_distance:NoitaFloat = field(default_factory=NoitaFloat)
    max_angle:NoitaFloat = field(default_factory=NoitaFloat)
    min_radius:NoitaInt = field(default_factory=NoitaInt)
    max_radius:NoitaInt = field(default_factory=NoitaInt)
    chunk_probability:NoitaFloat = field(default_factory=NoitaFloat)
    chunk_max_angle:NoitaFloat = field(default_factory=NoitaFloat)
    chunk_count:NoitaInt = field(default_factory=NoitaInt)
    chunk_material:NoitaInt = field(default_factory=NoitaInt)
    collapse_images:NoitaString = field(default_factory=NoitaString)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.probability.Read(s)
        self.max_durability.Read(s)
        self.max_distance.Read(s)
        self.max_angle.Read(s)
        self.min_radius.Read(s)
        self.max_radius.Read(s)
        self.chunk_probability.Read(s)
        self.chunk_max_angle.Read(s)
        self.chunk_count.Read(s)
        self.chunk_material.Read(s)
        self.collapse_images.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.probability.Write(s)
        self.max_durability.Write(s)
        self.max_distance.Write(s)
        self.max_angle.Write(s)
        self.min_radius.Write(s)
        self.max_radius.Write(s)
        self.chunk_probability.Write(s)
        self.chunk_max_angle.Write(s)
        self.chunk_count.Write(s)
        self.chunk_material.Write(s)
        self.collapse_images.Write(s)

